### PR TITLE
Qaqc issue 79

### PR DESCRIPTION
**Pull request by DavidGoldwasser**
I don't have any more changes planed for next measure gem release.

### CODE REVIEW

File: lib/measures/generic_qaqc/resources/check_eui_by_end_use.rb [hunk 1]
```diff
@@ -38,33 +38,46 @@ module OsLib_QAQC
...
-      use_old_gem_code = false
+    begin
+
+      # setup standard
       std = Standard.build(target_standard)
-    end
+      target_eui = std.model_find_target_eui(@model)
+
+      # gather building type for summary
+      bt_cz = std.model_get_building_climate_zone_and_building_type(@model)
```
Comment (mdahlhausen) [inline lib/measures/generic_qaqc/resources/check_eui_by_end_use.rb:71]: `model_get_building_climate_zone_and_building_type` no longer exists.  It's called `model_get_building_properties` now

File: lib/measures/generic_qaqc/resources/check_eui_reasonableness.rb [hunk 2]
```diff
@@ -38,33 +38,49 @@ module OsLib_QAQC
...
+      target_eui = std.model_find_target_eui(@model)
+
+      # gather building type for summary
+      bt_cz = std.model_get_building_climate_zone_and_building_type(@model)
+      building_type = bt_cz['building_type']
+      climate_zone = bt_cz['climate_zone']
+      prototype_prefix = "#{display_standard} #{building_type} #{climate_zone}"
+
+      # mapping to obuilding type to match space types
+      if building_type.include?("Office") then building_type = "Office" end
```
Comment (mdahlhausen) [inline lib/measures/generic_qaqc/resources/check_eui_reasonableness.rb:77]: building type enumerations

File: lib/measures/generic_qaqc/resources/check_eui_reasonableness.rb [hunk 3]
```diff
@@ -38,33 +38,49 @@ module OsLib_QAQC
...
-    if OpenStudio::VersionString.new(OpenStudio.openStudioVersion) < OpenStudio::VersionString.new('2.4.3')
-      use_old_gem_code = true
-    else
-      use_old_gem_code = false
+    begin
+
+      # setup standard
       std = Standard.build(target_standard)
-    end
+      target_eui = std.model_find_target_eui(@model)
```
Comment (mdahlhausen) [inline lib/measures/generic_qaqc/resources/check_eui_reasonableness.rb:68]: this won't work if the model doesn't have an enumerated building type set

File: lib/measures/generic_qaqc/resources/check_internal_loads.rb [hunk 4]
```diff
@@ -42,38 +42,48 @@ def check_internal_loads(category, target_standard, min_pass, max_pass, name_onl
...
-      use_old_gem_code = false
+    begin
+
+      # setup standard
       std = Standard.build(target_standard)
-    end
 
-    begin
+      # gather building type for summary
+      bt_cz = std.model_get_building_climate_zone_and_building_type(@model)
```
Comment (mdahlhausen) [inline lib/measures/generic_qaqc/resources/check_internal_loads.rb:73]: use `model_get_building_properties`

File: lib/measures/generic_qaqc/resources/check_schedules.rb [hunk 5]
```diff
@@ -42,60 +42,63 @@ def check_schedules(category, target_standard, min_pass, max_pass, name_only = f
...
-      use_old_gem_code = false
+    begin
+
+      # setup standard
       std = Standard.build(target_standard)
-    end
 
-    begin
+      # gather building type for summary
+      bt_cz = std.model_get_building_climate_zone_and_building_type(@model)
```
Comment (mdahlhausen) [inline lib/measures/generic_qaqc/resources/check_schedules.rb:69]: use `model_get_building_properties`

File: lib/measures/generic_qaqc/resources/check_schedules.rb [hunk 6]
```diff
@@ -255,21 +229,19 @@ def check_schedules(category, target_standard, min_pass, max_pass, name_only = f
...
-
+  def generate_load_insc_sch_check_attribute(target_hrs, load_inst, space_type, check_elems, min_pass, max_pass, target_standard)
     schedule_inst = nil
     inst_hrs = nil
 
+    # setup standard
+    std = Standard.build(target_standard)
+      
+    # gather building type for summary
+    bt_cz = std.model_get_building_climate_zone_and_building_type(@model)
```
Comment (mdahlhausen) [inline lib/measures/generic_qaqc/resources/check_schedules.rb:240]: use `model_get_building_properties`

**Review by mdahlhausen — CHANGES_REQUESTED** [review]
Most of it looks find.  In fixing the EUI lookups, I did change the `model_get_building_climate_zone_and_building_type` method to `model_get_building_properties` in openstudio-standards v0.2.16.  I'm guessing we are going to want to either 1) set OS 3.4 as the minimum version, or 2) initialize the `model_get_building_climate_zone_and_building_type` method in the resource file for this measure.  That's probably the best approach as there are many who still use OS v2.9 b/c of sketchup and app integration.

**Comment by DavidGoldwasser** [comment]
@mdahlhausen I'll need to have version of the measure that works with OpenStudio 3.3 to go with URBANopt that will be released shortly. Another URBANopt for 3.4 will come out a little later.

**Pull request merged**